tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb

```diff
@@ -0,0 +1,311 @@
+  ",
+  resource_props: {
+    'ingress_replication' => 'bgp',
+    'peer_ips'            => "['default']",
+    'suppress_arp'        => 'false',
+  },
+}
+
+tests['multicast_group'] = {
+  title_pattern:  'nve1 10000',
+  manifest_props: "
+    multicast_group => '224.1.1.1',
+    peer_ips        => 'default',
+    suppress_arp    => 'default'
+  ",
+  resource_props: {
+    'multicast_group' => '224.1.1.1',
+    'peer_ips'        => "['default']",
+    'suppress_arp'    => 'false',
+  },
+}
+
+tests['suppress_arp_true'] = {
+  title_pattern:  'nve1 10000',
+  manifest_props: "
+    suppress_arp => 'true'
+  ",
+  resource_props: {
+    'peer_ips'     => "['default']",
+    'suppress_arp' => 'true',
+  },
+}
+
+tests['suppress_arp_false'] = {
+  title_pattern:  'nve1 10000',
+  manifest_props: "
+    suppress_arp => 'false'
+  ",
+  resource_props: {
+    'peer_ips'     => "['default']",
+    'suppress_arp' => 'false',
+  },
+}
+
+# Note: In the current implementation assciate-vrf is a namevar.
+# It will be changed to a property in future. A correspoding test
+# will be added here.
+
+#################################################################
+# HELPER FUNCTIONS
+#################################################################
+
+# Full command string for puppet resource command
+def puppet_resource_cmd
+  cmd = PUPPET_BINPATH + 'resource cisco_vxlan_vtep_vni'
+  get_namespace_cmd(agent, cmd, options)
+end
+
+def build_manifest_cisco_vxlan_vtep_vni(tests, id)
+  if tests[id][:ensure] == :absent
+    state = 'ensure => absent,'
+    tests[id][:resource] = {}
+  else
+    state = 'ensure => present,'
+    manifest = tests[id][:manifest_props]
+    tests[id][:resource] = tests[id][:resource_props]
+  end
+
+  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
+  logger.debug("build_manifest_cisco_vxlan_vtep_vni :: title_pattern:\n" +
+               tests[id][:title_pattern])
+
+  # cisco_vxlan_vtep_vni needs cisco_vxlan_vtep as a prerequisite.
+  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
+  node 'default' {
+    cisco_vxlan_vtep {'nve1':
+      ensure => present,
+      host_reachability  => 'evpn',
+      shutdown           => 'false',
+    }
+    cisco_vxlan_vtep_vni { '#{tests[id][:title_pattern]}':
+      #{state}
+      #{manifest}
+    }
+  }
+EOF"
+end
+
+def test_harness_cisco_vxlan_vtep_vni(tests, id)
+  tests[id][:ensure] = :present if tests[id][:ensure].nil?
+  tests[id][:resource_cmd] = puppet_resource_cmd
+  tests[id][:desc] += " [ensure => #{tests[id][:ensure]}]"
+
+  build_manifest_cisco_vxlan_vtep_vni(tests, id)
+
+  test_manifest(tests, id)
+  test_resource(tests, id)
+  test_idempotence(tests, id)
+
+  tests[id][:ensure] = nil
+end
+
+#################################################################
+# TEST CASE EXECUTION
+#################################################################
+test_name "TestCase :: #{testheader}" do
+  #-------------
+  id = 'preclean'
```

Comment:
Its weird but using resource_absent_cleanup instead of 'preclean' test gives me a failure, which makes me content with 'preclean' test.

Failed Tests Cases:
Errored Tests Cases:
      Test Case cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb reported: #<Beaker::Host::CommandFailure: Host 'agent1' exited with 1 running:
     sudo ip netns exec management /opt/puppetlabs/bin/puppet resource cisco_vxlan_vtep_vni 'nve1 10000' ensure=absent
    Last 10 lines of output were:
        Notice: /Cisco_vxlan_vtep_vni[nve1 10000]/ensure: removed
        Error: Could not run: undefined method `default_peer_list' for nil:NilClass>
Skipped Tests Cases:
Pending Tests Cases: